### PR TITLE
Phase 3 slice 3: Mini player + cross-state audio

### LINKED ISSUE #32: Phase 3 slice 3: Mini player + cross-state audio
## Parent

#28

## What to build

Add a mini player anchored to the bottom of the viewport whenever the sheet is `closed` and audio is loaded. Tapping the mini player reopens the sheet to `peek`. This completes the cross-state audio UX: a visitor can drag the sheet down to dismiss while keeping the preview playing, and use the mini player to get the chart back.

Visibility rule: visible only when `currentTrack !== null` AND the sheet snap state is `closed`. Hidden otherwise.

`chart-screen.tsx` wires the sheet's snap state to the mini player's `sheetClosed` prop; the mini player's `onTap` callback fires a snap transition back to `peek`.

Phase 3 scope explicitly excludes scroll-to-active-track in the reopened sheet — deferred to Phase 5 alongside globe integration.

## Acceptance criteria

- [x] Mini player hidden when \`currentTrack === null\` (nothing loaded yet).
- [x] Mini player hidden when sheet snap state is \`peek\` or \`full\`.
- [x] Mini player visible at viewport bottom when sheet is \`closed\` AND \`currentTrack !== null\`.
- [x] Mini player shows artwork, track name, artist, and a play/pause toggle bound to the audio store.
- [x] Tapping the mini player transitions sheet snap from \`closed\` to \`peek\`.
- [x] Preview audio continues uninterrupted across sheet snap transitions (closed → peek → full and back), demonstrating the single-audio-instance contract.
- [x] \`mini-player.tsx\` RTL-tested for the three visibility branches (no current track / sheet not closed / both conditions met) and tap-callback firing.
- [x] CI green; manual \`pnpm dev\` verification of close-sheet-while-playing flow and tap-mini-to-reopen.

## Blocked by

- #31

### PR DESCRIPTION
**Pull request by taewanu**
## Summary

Slice 3 of Phase 3 (#28). Adds the mini player and the cross-state
audio behavior that ties the chart sheet, the mini, and the audio
store together. Visitors can drag the sheet down while a preview is
playing, the mini appears at the bottom of the viewport, tapping it
re-opens the sheet to peek with the active track scrolled into view,
and the next preview auto-plays when the current one finishes.

## Track row current-state indicator

- Current row exposes `data-state="playing"|"paused"` so a paused
  current track stays identifiable; rank flips to a play/pause icon in
  the same centered slot to avoid the visual jump from the number/SVG
  swap.
- Title and EQ adopt the sunrise accent; `data-paused` on the EQ
  freezes the animation via `animation-play-state: paused`.
- Hover-vs-current distinguished by `bg-atmos` (hover) and a brighter
  sunrise wash (current); the earlier inset orange ring + glow read
  as too strong.

## Chart sheet: list scrolls, drag-to-mini, scroll-to-current

- Track list scrolls inside the sheet without the sheet itself
  dragging — drag is restricted to handle + title via
  `useDragControls`.
- Dragging hard from full now lands directly at the mini (closed); the
  one-step-delta logic was capped at peek. `nextSnap` now projects
  final position + velocity.
- Sheet cannot be closed before a track is selected — prevents the
  dead-end where the page would be blank. `canClose` re-animates back
  via `useAnimationControls` because state-no-op didn't trigger motion.
- Re-opening the sheet auto-scrolls to the currently-playing track
  (rAF defer so Radix Portal mount completes; `block: start` at peek,
  `center` at full).
- Peek now scrolls end-to-end of the chart; `ol` `max-height` is
  capped to the visible peek area and relaxed during drag so the list
  grows in sync with the sheet.

## Mini player

- Bottom-anchored, visible only when `currentTrack !== null` AND sheet
  is `closed`.
- Left region (artwork + title + artist) re-opens the sheet on tap.
- Right toggle bound to the same audio store; title and EQ mirror the
  track row's sunrise treatment so the two surfaces feel like one
  player.

## Chart-screen wiring + audio UX

- `AudioStoreProvider` wraps an inner `ChartScreenInner` so the sheet
  and mini share one audio store and the snap state lives below the
  provider. Selectors narrowed to primitives (`hasCurrentTrack`,
  `currentTrackRank`) to keep this component out of the re-render
  path on track switches.
- Spacebar anywhere on the page toggles play/pause for the current
  track; skips input/textarea/contenteditable so search fields aren't
  affected.
- 30s preview end auto-advances to the next track that has a preview;
  chart end stops playback. Audio store gains `endedSignal` (a counter
  the consumer subscribes to) and `useAudioStoreApi` exposes the
  imperative store handle so key handlers + effects can read state
  without re-subscribing.

## Notable decisions

- Visible next/prev buttons and keyboard nav for them deferred to a
  Phase 4-5 polish pass; auto-advance is the V1 substitute that keeps
  the chart flowing without extra UI.
- E2E browser tests explicitly deferred to Phase 5 entry. RTL + jsdom
  cannot exercise drag/scroll/touch behaviors that slice 3 surfaced;
  manual verify is the Phase 3-4 safety net. Saved as project memory.

## Test plan

- [x] lint / typecheck / test green (111 tests across 18 files)
- [x] Manual \`pnpm dev\` golden path: load → play → drag closed → mini
      visible → tap → peek → drag full → drag closed direct → mini
- [x] Edge cases: dead-end clamp before any track plays; auto-advance
      stops at last track; spacebar skipped in inputs; peek scrolls
      end-to-end; rank ↔ icon swap centered with no jump; paused EQ
      stays static
- [x] CI green
- [x] CodeRabbit

Closes #32

## Summary by CodeRabbit

* **New Features**
  * Spacebar now toggles audio playback (when not in text input)
  * Mini player displays current track details when chart is minimized
  * Auto-advances to next track when current track finishes

* **Improvements**
  * Enhanced track row scrolling and positioning
  * Refined play/pause icon display in track listings
  * Updated equalizer animation behavior

[Review Change Stack](https://app.coderabbit.ai/change-stack/taewanu/sounds-abroad/pull/40?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/app/chart-screen.tsx [hunk 1]
```diff
@@ -38,16 +43,75 @@ export function ChartScreen({ charts, rng = Math.random }: ChartScreenProps) {
...
+        (target.tagName === "INPUT" ||
+          target.tagName === "TEXTAREA" ||
+          target.isContentEditable)
+      ) {
+        return;
+      }
+      const { currentTrack, toggle } = audioStore.getState();
+      if (currentTrack === null) return;
+      e.preventDefault();
+      toggle(currentTrack);
```
[the remark this answers is not in the document]
Comment (taewanu) [reply src/app/chart-screen.tsx:85]: Applied in 9c2ead0 — Space now also skips on focused interactive elements (button, a[href], SELECT, role=button/menuitem/switch/checkbox) and on Space with modifier keys.

File: src/components/chart-sheet/sheet.test.tsx [hunk 2]
```diff
@@ -65,4 +65,99 @@ describe("ChartSheet", () => {
 
     expect(onSnapChange).toHaveBeenCalledWith("peek");
   });
+
+  test("scrolls currentTrack into view when sheet transitions from closed", async () => {
+    const scrollIntoViewMock = vi.fn();
+    Element.prototype.scrollIntoView = scrollIntoViewMock;
```
[the remark this answers is not in the document]
Comment (taewanu) [reply src/components/chart-sheet/sheet.test.tsx:77]: Applied in 9bf2236 — `afterEach` restores `Element.prototype.scrollIntoView` from the captured original.
